mvc/app.js

```diff
@@ -0,0 +1,19 @@
+const express = require("express");
+const app = express();
+const { getAllTopics } = require("../mvc/controller");
+const { handle404Errors, handleServerErrors } = require("../mvc/middleware");
+
+app.use(express.json());
+
+app.get("/api/healthcheck", (req, res) => {
+  res.status(200).send({msg: "healthcheck passed"});
+});
+
+app.get("/api/topics", getAllTopics);
+
+app.all("/*", handle404Errors);
+
+app.all("/*", handleServerErrors);
```

Comment:
You can't have to pieces of middleware dealing with the same endpoint. When you have it set up like this the first piece of middleware that deals with that endpoint will take precedence. Currently we would never get to your handleServerErrors function. 

That should be invoked using the .use() method, not the .all() method.